samples/event_manager_profiler/src/modules/module_a.c

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+
+#include "five_sec_event.h"
+#include "one_sec_event.h"
+#include "config_event.h"
+
+#define MODULE_A_THREAD_STACK_SIZE 800
+#define MODULE_A_THREAD_PRIORITY 1
+#define MODULE_A_THREAD_SLEEP 1000
+#define MODULE_A_WORK_SIMULATE_SLEEP 100
+#define MODULE module_a
+
+
+static K_THREAD_STACK_DEFINE(module_a_thread_stack,
+			     MODULE_A_THREAD_STACK_SIZE);
+static struct k_thread module_a_thread;
+
+static uint8_t cnt;
+
+void send_one_sec_event(void)
+{
+	struct one_sec_event *event = new_one_sec_event();
+
+	event->five_sec_timer = cnt;
+
+	EVENT_SUBMIT(event);
+}
+
+static void module_a_thread_fn(void)
+{
+
+	while (true) {
+		cnt++;
+		send_one_sec_event();
+		k_sleep(K_MSEC(MODULE_A_THREAD_SLEEP));
+
+	}
+}
+
+static void init(void)
+{
+	k_thread_create(&module_a_thread,
+			module_a_thread_stack,
+			MODULE_A_THREAD_STACK_SIZE,
+			(k_thread_entry_t)module_a_thread_fn,
+			NULL, NULL, NULL,
+			MODULE_A_THREAD_PRIORITY,
+			0, K_NO_WAIT);
+}
+
+
+static bool event_handler(const struct event_header *eh)
+{
+	if (is_config_event(eh)) {
+		init();
+		return false;
+	}
+
+	if (is_five_sec_event(eh)) {
+		cnt = 0;
+		k_msleep(MODULE_A_WORK_SIMULATE_SLEEP);
```

Comment:
I think busy wait could be better to simulate work/calculations. Sleep gives context so CPU would not as if it was performing operations in given thread.